pydantic_factories/constraints/date.py

```diff
@@ -0,0 +1,31 @@
+from datetime import date, timedelta
+from typing import TYPE_CHECKING, Optional, cast
+
+if TYPE_CHECKING:
+    from faker import Faker
+    from pydantic import ConstrainedDate
+
+
+def handle_constrained_date(constrained_date: "ConstrainedDate", faker: "Faker") -> date:
+    """
+    Generates a date value fulfilling the expected constraints.
+    Args:
+        constrained_date:
+        faker:
+
+    Returns:
+
+    """
+    start_date: Optional[date] = date.today() - timedelta(days=100)
+    if constrained_date.ge:
+        start_date = constrained_date.ge
+    elif constrained_date.gt:
+        start_date = constrained_date.gt + timedelta(days=1)
+
+    end_date: Optional[date] = date.today() + timedelta(days=100)
+    if constrained_date.le:
+        end_date = constrained_date.le
+    elif constrained_date.lt:
+        end_date = constrained_date.lt - timedelta(days=1)
+
+    return cast("date", faker.date_between(start_date=start_date, end_date=end_date))
```

Comment:
As far as I remember date_between output won't include end_date.

Reply:
@Goldziher strange, I can not find docs but I receive weird results when I do manual testing

It is so improbable to see such distribution

```
In [176]: from collections import defaultdict

In [177]: seen = defaultdict(int)

In [178]: fake = Faker()

In [179]: for _ in range(100000):
     ...:     seen[fake.date_between(start_date='-1d', end_date='today')] += 1
     ...: 

In [180]: seen
Out[180]: 
defaultdict(int,
            {datetime.date(2022, 10, 13): 99999,
             datetime.date(2022, 10, 14): 1})
```

Anyway, it seems we can leave it as is and just keep in mind that there might be an issue with Faker itself.

Reply:
not according to the code: https://github.com/joke2k/faker/blob/master/faker/providers/date_time/__init__.py#L2101